src/protocols/secure_channel/PASESession.cpp

```diff
@@ -707,64 +701,74 @@ CHIP_ERROR PASESession::HandleMsg3(const System::PacketBufferHandle & msg)
 
 void PASESession::SendErrorMsg(Spake2pErrorType errorCode)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
-
     System::PacketBufferHandle msg;
     uint16_t msglen        = sizeof(Spake2pErrorMsg);
     Spake2pErrorMsg * pMsg = nullptr;
 
     msg = System::PacketBufferHandle::New(msglen);
-    VerifyOrExit(!msg.IsNull(), err = CHIP_SYSTEM_ERROR_NO_MEMORY);
+    VerifyOrReturn(!msg.IsNull(), ChipLogError(SecureChannel, "Failed to allocate error message"));
 
     pMsg        = reinterpret_cast<Spake2pErrorMsg *>(msg->Start());
     pMsg->error = errorCode;
 
     msg->SetDataLength(msglen);
 
-    err = mExchangeCtxt->SendMessage(Protocols::SecureChannel::MsgType::PASE_Spake2pError, std::move(msg));
-    SuccessOrExit(err);
-
-exit:
-    Clear();
+    VerifyOrReturn(mExchangeCtxt->SendMessage(Protocols::SecureChannel::MsgType::PASE_Spake2pError, std::move(msg)) ==
+                       CHIP_NO_ERROR,
+                   ChipLogError(SecureChannel, "Failed to send error message"));
 }
 
-void PASESession::HandleErrorMsg(const System::PacketBufferHandle & msg)
+CHIP_ERROR PASESession::HandleErrorMsg(const System::PacketBufferHandle & msg)
 {
-    // Request message processing
-    const uint8_t * buf    = msg->Start();
-    size_t buflen          = msg->DataLength();
-    Spake2pErrorMsg * pMsg = nullptr;
+    ReturnErrorCodeIf(msg->Start() == nullptr || msg->DataLength() != sizeof(Spake2pErrorMsg), CHIP_ERROR_MESSAGE_INCOMPLETE);
 
-    VerifyOrExit(buf != nullptr, ChipLogError(Ble, "Null error msg received during pairing"));
-    VerifyOrExit(buflen == sizeof(Spake2pErrorMsg), ChipLogError(Ble, "Error msg with incorrect length received during pairing"));
+    static_assert(
+        sizeof(Spake2pErrorMsg) == sizeof(uint8_t),
+        "Assuming size of Spake2pErrorMsg message is 1 octet, so that endian-ness conversion and memory alignment is not needed");
 
-    pMsg = reinterpret_cast<Spake2pErrorMsg *>(msg->Start());
-    ChipLogError(Ble, "Received error (%d) during pairing process", pMsg->error);
+    Spake2pErrorMsg * pMsg = reinterpret_cast<Spake2pErrorMsg *>(msg->Start());
+    ChipLogError(SecureChannel, "Received error during pairing process. %s", ErrorStr(pMsg->error));
 
-    mDelegate->OnSessionEstablishmentError(pMsg->error);
-
-exit:
-    Clear();
+    return pMsg->error;
```

Comment:
This part doesn't make sense to me.  pMsg->error is a Spake2pErrorType, not a CHIP_ERROR...